### ISSUE TITLE
youtube-dlg update error

### ISSUE THREAD

**Issue by ip018** [MrS0m30n3/youtube-dl-gui#484]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer **honestly**
- Put an `x` into all the boxes [ ] relevant to your issue (like that [x])
- Use *Preview* tab to see how your issue will actually look like

### WARNING
All invalid issues will be rejected!!

---

### Before going further

- If your problem is a bug with **youtube-dl** or a request for new site support please report it [here](https://github.com/rg3/youtube-dl/issues)

- Make sure you are using the *latest* **youtube-dl-gui** version (Click the `Settings` icon and then `About` to view the current version)

- Make sure you are using the *latest* **youtube-dl** version (Click the `Settings` icon and then `Update` to update to the latest **youtube-dl** version)

- Make sure you searched the bugtracker for similar issues **including closed ones**

- Make sure to read the [FAQs](https://github.com/MrS0m30n3/youtube-dl-gui/blob/master/docs/faqs.md) file

  - [x] **I think** my problem is **NOT** with **youtube-dl**
  - [x] I've **verified** and **i assure** that I'm running youtube-dl-gui **0.4**
  - [x] **I assure** that i am using the latest version of **youtube-dl**
  - [x] [Searched](https://github.com/MrS0m30n3/youtube-dl-gui/issues) bugtracker
  - [x] I've read the FAQs file

---

### What is the purpose of your *issue*?

- [x] Bug report
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

Please remove any sections between (---) if they are not related to your issue

---

#### If the problem occurs when downloading a URL please provide the full verbose output as follows:

1. Restart **youtube-dl-gui**
1. Go to `Options > Extra` tab
2. Enable **Debug youtube-dl**
3. Go to `Options > Advanced` tab and **Clear** your log content
4. Try to download the URL
5. Copy the **whole** log content and insert it between the ``` part below

```
delete me and insert your log content here
```

#### What operating system do you use ?

Windows 7 Home Premium 64 bit

#### List of actions to perform to reproduce the problem:

  1. Open youtube-dlg
  2. Open settings
  3. Select "update"
  
#### What is the expected behaviour ?

It should say "downloading latest youtube-dl. please wait...", and then after a few seconds say something like "youtube-dl downloaded successfully.".

#### What happens instead ?

It says "downloading latest youtube-dl. please wait...", but then after a few seconds it says "youtube-dl download failed [http error 403: forbidden]".

![youtube-dlg 1](https://user-images.githubusercontent.com/35790889/97813059-aae61200-1c53-11eb-8eaa-aef2a5be66ce.jpg)

![youtube-dlg 2](https://user-images.githubusercontent.com/35790889/97813064-b0dbf300-1c53-11eb-99e5-068f29f74ba7.jpg)

---

### Feature request (request for a new functionality)

Please make sure that the requested feature is **NOT** already in the [TODO](https://github.com/MrS0m30n3/youtube-dl-gui/blob/master/TODO) list

- [x] I've **verified** and **i assure** that my requested feature is **NOT** in the TODO list

#### What operating system do you use ?

Windows 7 Home Premium 64 bit

---

When you go to youtube-dlg's settings and select "update", it says "downloading latest youtube-dl. please wait...", but then after a few seconds it says "youtube-dl download failed [http error 403: forbidden]". This issue never used to happen before, and it always used to just download the update successfully and say "downloading latest youtube-dl. please wait...", and then after a few seconds say something like "youtube-dl downloaded successfully." without any issues. Nothing has changed with the program installation or settings. Is there a way to fix that? Thanks.

**Comment by Technetium1** [comment]
This https://github.com/MrS0m30n3/youtube-dl-gui/blob/c5c18e55cb7e04fb6d6d8e64024a7dbac1f6b431/youtube_dl_gui/updatemanager.py#L48
needs to be changed to ```https://youtube-dl.org/downloads/latest/youtube-dl```

If you're on windows you can download `youtube-dl.exe` and place it in `%AppData%\youtube-dlg`

**Comment by UltraBlackLinux** [comment]
Does not work for me, still 403

**Comment by lunarbunny** [comment]
> This
> 
> https://github.com/MrS0m30n3/youtube-dl-gui/blob/c5c18e55cb7e04fb6d6d8e64024a7dbac1f6b431/youtube_dl_gui/updatemanager.py#L48
> 
> needs to be changed to `https://youtube-dl.org/downloads/latest/youtube-dl.exe`
> 
> If you're on windows you can download `youtube-dl.exe` and place it in `%AppData%\youtube-dlg`

The equivalent path for linux would be `~/.config/youtube-dlg/`

I simply overwrote it using the latest version installed from `pip`:
`cat ~/.local/bin/youtube-dl > .config/youtube-dlg/youtube-dl`

**Comment by RedBreloom** [comment]
> 
> 
> This
> 
> https://github.com/MrS0m30n3/youtube-dl-gui/blob/c5c18e55cb7e04fb6d6d8e64024a7dbac1f6b431/youtube_dl_gui/updatemanager.py#L48
> 
> needs to be changed to `https://youtube-dl.org/downloads/latest/youtube-dl.exe`
> 
> If you're on windows you can download `youtube-dl.exe` and place it in `%AppData%\youtube-dlg`

I have the windows exe version so I tried to replace the .exe in the specified directory, but it doesn't appear to have done anything for me either. Do I need to update something else?

**Comment by Technetium1** [comment]
@RedBreloom Edited my comment, it shouldn't have `.exe`

**Comment by RedBreloom** [comment]
> 
> 
> @RedBreloom Edited my comment, it shouldn't have `.exe`

I am using the windows executable version of this app so I am not sure how to access the code directly and trying to replace the youtube-dl.exe in appdata it doesn't seem to fix this issue. 
I also realized that this error post is not the same as the one i have. But another person who had the same problem as I was directed to this post so I figured i'd ask here as well. i'm not really sure what to do otherwise.
#https://github.com/MrS0m30n3/youtube-dl-gui/issues/488
for reference

**Comment by Technetium1** [comment]
@RedBreloom make sure youtube-dl isn't in your path somewhere else already, if all else fails remove it all and try again. You should probably make another issue if this persists.

**Comment by tronics** [comment]
What you need on Windows is this library: 
https://www.microsoft.com/en-US/download/details.aspx?id=5555
This is what makes it work on a freshly installed Windows 10. 

All of the above commentary is not necessary since the library is back on Github despite interventions from Google.

**Comment by erhan-** [comment]
> 
> 
> What you need on Windows is this library:
> https://www.microsoft.com/en-US/download/details.aspx?id=5555
> This is what makes it work on a freshly installed Windows 10.
> 
> All of the above commentary is not necessary since the library is back on Github despite interventions from Google.

Yep this is what I do on all systems of friends where I install youtube-dl-gui. I always had to search inside the closed Issues for the link. 

Just tested on a freshly installed system. It works as usual. Too bad this is not more visible.

**Comment by RedBreloom** [comment]
As a note: I already had a more recent version of that redis installed on my computer. So I don't think that mattered for my case.. It doesn't seem like this program is being maintained anymore anyway, so I've been using tartube for now.
